Handle slow/dead network connection on session rename more gracefully

2025-10-10 12:02:17,957 [pool-7-thread-5] ERROR ContextManager.lambda$renameSessionAsync$79:2273 - Error renaming Session 0d9297f2-b79d-452f-816d-f54ee4da72e6
java.util.concurrent.TimeoutException: null
	at java.util.concurrent.FutureTask.get(FutureTask.java:204) ~[?:?]
	at javax.swing.SwingWorker.get(SwingWorker.java:624) ~[?:?]
	at io.github.jbellis.brokk.ContextManager.lambda$renameSessionAsync$79(ContextManager.java:2269) ~[brokk-0.15.0.4-9-g26167408a.jar:?]
	at io.github.jbellis.brokk.ContextManager.lambda$submitBackgroundTask$53(ContextManager.java:1601) ~[brokk-0.15.0.4-9-g26167408a.jar:?]
	at io.github.jbellis.brokk.ContextManager.lambda$submitBackgroundTask$52(ContextManager.java:1566) ~[brokk-0.15.0.4-9-g26167408a.jar:?]
	at io.github.jbellis.brokk.util.LoggingExecutorService.lambda$wrap$6(LoggingExecutorService.java:102) ~[brokk-0.15.0.4-9-g26167408a.jar:?]
	at io.github.jbellis.brokk.util.LoggingExecutorService.lambda$submit$0(LoggingExecutorService.java:34) ~[brokk-0.15.0.4-9-g26167408a.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.lang.Thread.run(Thread.java:1575) [?:?]
2025-10-10 12:02:17,958 [pool-7-thread-5] ERROR ContextManager.lambda$createLoggingExecutorService$2:120 - Uncaught exception in executor
java.lang.RuntimeException: java.util.concurrent.TimeoutException
	at io.github.jbellis.brokk.ContextManager.lambda$renameSessionAsync$79(ContextManager.java:2274) ~[brokk-0.15.0.4-9-g26167408a.jar:?]
	at io.github.jbellis.brokk.ContextManager.lambda$submitBackgroundTask$53(ContextManager.java:1601) ~[brokk-0.15.0.4-9-g26167408a.jar:?]
	at io.github.jbellis.brokk.ContextManager.lambda$submitBackgroundTask$52(ContextManager.java:1566) ~[brokk-0.15.0.4-9-g26167408a.jar:?]
	at io.github.jbellis.brokk.util.LoggingExecutorService.lambda$wrap$6(LoggingExecutorService.java:102) ~[brokk-0.15.0.4-9-g26167408a.jar:?]
	at io.github.jbellis.brokk.util.LoggingExecutorService.lambda$submit$0(LoggingExecutorService.java:34) ~[brokk-0.15.0.4-9-g26167408a.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.lang.Thread.run(Thread.java:1575) [?:?]
Caused by: java.util.concurrent.TimeoutException
	at java.util.concurrent.FutureTask.get(FutureTask.java:204) ~[?:?]
	at javax.swing.SwingWorker.get(SwingWorker.java:624) ~[?:?]
	at io.github.jbellis.brokk.ContextManager.lambda$renameSessionAsync$79(ContextManager.java:2269) ~[brokk-0.15.0.4-9-g26167408a.jar:?]
	... 9 more

--- a/app/src/main/java/io/github/jbellis/brokk/ContextManager.java
+++ b/app/src/main/java/io/github/jbellis/brokk/ContextManager.java
@@ -2270,8 +2270,7 @@ public class ContextManager implements IContextManager, AutoCloseable {
                 project.getSessionManager().renameSession(sessionId, newName);
                 logger.debug("Renamed session {} to {}", sessionId, newName);
             } catch (Exception e) {
-                logger.error("Error renaming Session {}", sessionId, e);
-                throw new RuntimeException(e);
+                logger.warn("Error renaming Session {}", sessionId, e);
             }
         });
     }

